src/main/scala/sigmastate/utils/ByteReader.scala

```diff
@@ -122,6 +130,15 @@ class ByteBufferReader(buf: ByteBuffer) extends ByteReader {
   }
 
   @inline override def getBytes(size: Int): Array[Byte] = buf.getBytes(size)
+
+  @inline override def getBits(size: Int): Array[Boolean] = {
+    if (size == 0) return Array[Boolean]()
+    val bitSet = BitSet.valueOf(buf.getBytes((size + 7) / 8))
+    (0 until size)
+      .map(i => bitSet.get(i))
```

Comment:
@greenhat I think we should forbid functional style operations in performance critical code. Please change to code that avoid unnecessary intermediate allocations.

Reply:
I got carried away. Will do. Should we consider using Scalan? :)

Reply:
not yet 😄